docs/state-diff/README.md

```diff
@@ -0,0 +1,185 @@
+
+**A note on interfaces**: One of the goals of this technique is to isolate the verified bytecode from the sourcecode producing it. Therefore `Counter` here is just an interface of the actual `Counter` contract. The interface can be found [here](test/kontrol/proofs/utils/Interfaces.sol). To see more clearly how this is brought together, see the test [here](test/kontrol/proofs/Counter.k.sol).
+
+Note that [our test](test/kontrol/proofs/Counter.k.sol) is named `prove_multiple_counters`. The reason for not using `test_` is that, since we're not including the `recreateDeployment` function in the `setUp`, running that test with `forge` will not be successful, since the state updates haven't been loaded.
+
+However, we don't need to run the `recreateDeployment` function in Kontrol! Let's see how we can execute `prove_multiple_counters`.
+
+## ⚙️ Run Your Proofs ⚙️
+
+At this point we have a test that says things about addresses, but no information about the addresses is actually present in the contract! That is, we have following `setUp` function that just stores a bunch of addresses:
+```solidity
+Counter[] public counters;
+
+function setUp() public {
+    counters.push(Counter(address(counter0Address)));
+    counters.push(Counter(address(counter1Address)));
+    counters.push(Counter(address(counter2Address)));
+    counters.push(Counter(address(counter3Address)));
+    counters.push(Counter(address(counter4Address)));
+    counters.push(Counter(address(counter5Address)));
+    counters.push(Counter(address(counter6Address)));
+    counters.push(Counter(address(counter7Address)));
+    counters.push(Counter(address(counter8Address)));
+    counters.push(Counter(address(counter9Address)));
+}
+```
+And the following tests that asserts some property for each of the stored addresses:
+```solidity
+function prove_multiple_counters() public {
+        for(uint256 i; i <= 9; ++i){
+            require(counters[i].number() == i, "Bad number");
+        }
+    }
+```
+All that is left is to tell Kontrol which state updates need to be loaded before executing `prove_multiple_counters`. This is done via the `--init-node-from` flag of the `kontrol prove` command. The `--init-node-from` flag expects a JSON file containing the state updates and will load them to the initial state of the proof. This will make Kontrol aware of all the changes that occurred during the execution of the original `setUp` function, with none of the computation.
+
+Hence, to successfully execute the above function in Kontrol we'll have to execute:
+1. `FOUNDRY_PROFILE=kprove kontrol build`
+2. `FOUNDRY_PROFILE=kprove kontrol prove --match-test prove_multiple_counters --init-node-from state-diff/StateDiff.json`
+Note that running this in the context of the `kprove` profile is crucial, since it points to the isolated folder that will contain all the necessary bytecode.
```

Comment:
Maybe you can show the result? Like the `PROOF PASSED: ...` log? I think we can use some closure in this tutorial :)